docs/learn/smart-contract-developers/getting-started.md

```diff
@@ -4,132 +4,342 @@ sidebar_position: 1
 
-## Get testnet LYXt
+:::danger
 
-To pay for the deployment fees, you need LYXt. You can request some from the [LUKSO Testnet faucet](https://faucet.testnet.lukso.network/)
+Never share your private key that you will put into the `.env` file!
 
-## Deploy your contracts on the LUKSO Testnet
+:::
 
-:::info
+:::success
 
-By default, the deployment will be to your local network. If you want to deploy to the LUKSO Testnet, you will need to add the LUKSO Testnet network in your `hardhat.config.ts`.
+To pay for the deployment fees of the accounts, you will need LYXt or LYX. For the testnet, you can request LYXt from the [LUKSO Testnet faucet](https://faucet.testnet.lukso.network/)
 
 :::
 
-```js title="hardhat.config.ts"
-// ...
-import { NetworkUserConfig } from 'hardhat/types';
+We now have a base Hardhat setup that we can use to develop and deploy our smart contracts.
 
-import * as dotenv from 'dotenv';
-dotenv.config();
+## Configure contracts and networks
 
-// ...
+By default, the deployment will be to your local network and default Solidity version. If you want to deploy and verify your smart contracts on LUKSO, you will have to
 
-function getTestnetChainConfig(): NetworkUserConfig {
-  const config: NetworkUserConfig = {
-    url: 'https://rpc.testnet.lukso.network',
-    chainId: 4201,
-  };
+- specify `solidity` compiler settings, so LSP presets match your contract version and EVM requirements
+- add the parameters of the `networks` and their related private keys
+- define the `etherscan` APIs to verify contracts on the networks
 
-  if (process.env.PRIVATE_KEY !== undefined) {
-    config['accounts'] = [process.env.PRIVATE_KEY];
-  }
+```js title="hardhat.config.ts"
+import { HardhatUserConfig } from 'hardhat/config';
+import { config as LoadEnv } from 'dotenv';
+import '@nomicfoundation/hardhat-toolbox';
 
-  return config;
-}
+LoadEnv();
 
-// Edit the default config object so it matches this one:
 const config: HardhatUserConfig = {
   solidity: {
+    // Default compiler version for all contracts
     version: '0.8.19',
     settings: {
       optimizer: {
         enabled: true,
-        runs: 200,
+        runs: 1000,
       },
     },
   },
-  typechain: {
-    outDir: 'typechain-types',
-    target: 'ethers-v6',
-  },
   networks: {
-    luksoTestnet: getTestnetChainConfig(),
+    luksoTestnet: {
+      url: 'https://rpc.testnet.lukso.gateway.fm',
+      chainId: 4201,
+      accounts: process.env.PRIVATE_KEY ? [process.env.PRIVATE_KEY] : [],
+    },
+    luksoMainnet: {
+      url: 'https://rpc.lukso.gateway.fm',
+      chainId: 42,
+      accounts: process.env.PRIVATE_KEY ? [process.env.PRIVATE_KEY] : [],
+    },
+  },
+  etherscan: {
+    apiKey: 'no-api-key-needed',
```

Comment:
maybe

```suggestion
    // apiKey: '', - No API key needed for LUKSO explorer
```

Reply:
The entry (`apiKey`) is necessary to run the verify command. If it is not needed, you can put whatever you want